functions/slack/search.go

```diff
@@ -71,27 +94,85 @@ func KGSearch(w http.ResponseWriter, r *http.Request) {
 
 // [END functions_slack_search]
 
+// [START functions_slack_request]
+func makeSearchRequest(query string) (*Message, error) {
+	res, err := entitiesService.Search().Query(query).Limit(1).Do()
+	if err != nil {
+		return nil, fmt.Errorf("Do: %v", err)
+	}
+	return formatSlackMessage(query, res)
+}
+
+// [END functions_slack_request]
+
 // [START functions_verify_webhook]
-func verifyWebHook(form url.Values) error {
-	t := form.Get("token")
-	if len(t) == 0 {
-		return fmt.Errorf("empty form token")
+
+// VeryfyWebhook now uses signature verification instead of tokens.
+// see https://api.slack.com/docs/verifying-requests-from-slack
+func verifyWebHook(r *http.Request, secret string) error {
+	result, err := verifyRequestSignature(r, secret)
```

Comment:
Consider inlining `verifyRequestSignature`, since this function and that essentially do the same thing and inlining would remove one layer of indirection.